torch/csrc/jit/script/compiler.cpp

```diff
@@ -2309,7 +2308,7 @@ struct to_ir {
           elem_type = values.at(0)->type();
         }
         for (auto v : values) {
-          if (v->type() != elem_type) {
+          if (*v->type() != *elem_type)  {
```

Comment:
what's wrong with here? is it because some `v` might be optional?

Reply:
It's working now. Previously it would fail for something like int[] != int[]